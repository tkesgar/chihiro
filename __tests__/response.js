/* eslint-env jest */
const {errorToResponse, resultToResponse} = require('../lib/response')

describe('errorToResponse', () => {
  test('no id', () => {
    const error = {}
    expect(errorToResponse(error)).toBe(undefined)
  })

  test('empty object', () => {
    const error = {}
    expect(() => errorToResponse(error, 'foo')).toThrow()
  })

  test('no code', () => {
    const error = {
      message: 'An error has occured'
    }
    expect(() => errorToResponse(error, 'foo')).toThrow()
  })

  test('no message', () => {
    const error = {
      code: 100
    }
    expect(() => errorToResponse(error, 'foo')).toThrow()
  })

  test('with id', () => {
    const error = {
      code: 100,
      message: 'An error has occured'
    }

    expect(errorToResponse(error, 'foo')).toMatchSnapshot()
  })

  test('with data', () => {
    const error = {
      code: 100,
      message: 'An error has occured',
      data: {
        key: 'abc',
        value: 123
      }
    }

    expect(errorToResponse(error, 'foo')).toMatchSnapshot()
  })

  test('with falsy data', () => {
    const error = {
      code: 100,
      message: 'An error has occured',
      data: null
    }

    expect(errorToResponse(error, 'foo')).toMatchSnapshot()
  })
})

describe('resultToResponse', () => {
  test('no id', () => {
    const result = {
      key: 'abc',
      value: 123
    }

    expect(resultToResponse(result)).toBe(undefined)
  })

  test('with id', () => {
    const result = {
      key: 'abc',
      value: 123
    }

    expect(resultToResponse(result, 'foo')).toMatchSnapshot()
  })
})
