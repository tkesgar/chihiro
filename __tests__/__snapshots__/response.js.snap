// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`errorToResponse with data 1`] = `
Object {
  "error": Object {
    "code": 100,
    "data": Object {
      "key": "abc",
      "value": 123,
    },
    "message": "An error has occured",
  },
  "id": "foo",
  "jsonrpc": "2.0",
}
`;

exports[`errorToResponse with falsy data 1`] = `
Object {
  "error": Object {
    "code": 100,
    "data": null,
    "message": "An error has occured",
  },
  "id": "foo",
  "jsonrpc": "2.0",
}
`;

exports[`errorToResponse with id 1`] = `
Object {
  "error": Object {
    "code": 100,
    "message": "An error has occured",
  },
  "id": "foo",
  "jsonrpc": "2.0",
}
`;

exports[`resultToResponse with id 1`] = `
Object {
  "id": "foo",
  "jsonrpc": "2.0",
  "result": Object {
    "key": "abc",
    "value": 123,
  },
}
`;
